lua/cfc_http_restrictions/wrap_functions.lua

```diff
@@ -107,6 +107,45 @@ local function wrapPlayURL()
     end
 end
 
+
+local function wrapHTMLPanel(panelName)
+    print( "wrapped SetHTML and OpenURL for "..panelName )
+    funcName = function(functionName)
+        return "_"..panelName.."_"..functionName
+    end
+
+    _G[funcName("SetHTML")] =  _G[funcName("SetHTML")] or vgui.GetControlTable(panelName).SetHTML
+    _G[funcName("OpenURL")] =  _G[funcName("OpenURL")] or vgui.GetControlTable(panelName).OpenURL
+    
+    vgui.GetControlTable(panelName).SetHTML = function( self, html, ... )
+
+        local isAllowed, url = CFCHTTP.isHTMLAllowed( html ) 
+
+        local stack = string.Split( debug.traceback(), "\n" )
+        logRequest( "GET", url, stack[3], isAllowed )
+
+        if not isAllowed then
+            html = [[<h1> BLOCKED </h1>]] 
+        end
+    
+        _G[funcName("SetHTML")]( self, html, ... )
+    end
+    
+    vgui.GetControlTable(panelName).OpenURL = function( self, url, ... )
+        local isAllowed = CFCHTTP.isAllowed( url )
+        local stack = string.Split( debug.traceback(), "\n" )
+        logRequest( "GET", url, stack[3], isAllowed )
+        if not isAllowed then return end
+
+        _G[funcName("OpenURL")]( self, url, ... )
+    end
+    
+end
+
+wrapHTMLPanel("DHTML")
+wrapHTMLPanel("DPanel")
+wrapHTMLPanel("DMediaPlayerHTML")
```

Comment:
When I tested locally, these elements weren't in the vgui control table until later. I just picked `Initialize` at random and it worked, but I'm sure post-gamemode would work too

```suggestion
hook.Add( "Initialize", "CFC_HttpWhitelist_WrapHTML", function()
    wrapHTMLPanel("DHTML")
    wrapHTMLPanel("DPanel")
    wrapHTMLPanel("DMediaPlayerHTML")
end )
```